lib/merchant.rb

```diff
@@ -22,10 +24,21 @@ def invoices
     @invoices ||= @repository.find_invoices(id)
   end
 
-  def revenue
-    successful_invoice_items =  successful_invoices.map { |i| i.invoice_items }.flatten
-    successful_invoice_items.reduce(0) do |revenue, invoice_item|
-      revenue + (invoice_item.quantity * invoice_item.unit_price)
+  def revenue(date=nil)
+    if date.is_a?(Date)
+      invoices_for_given_date = invoices.select { |i| Date.parse(i.created_at) == date }
+      successful_transactions ||= invoices_for_given_date.map { |i| i.transactions }.flatten
+      successful_transactions.select! { |t| t.result == 'success' }
+      successful_invoices_for_given_date = successful_transactions.map { |t| t.invoice }
```

Comment:
Line is too long. [88/80]<br>Pass `&:invoice` as an argument to `map` instead of a block.